### PR TITLE
remove mail handling

### PR DESCRIPTION
**Pull request by flga**
please note that in legacy-queue.js line 1 the bot uid is currently hardcoded.

### CODE REVIEW

File: backend/src/site/legacy-queue.js [hunk 1]
```diff
@@ -1,34 +1,401 @@
...
-	user: 'leechba1@gmail.com',
-	pass: 'shadowsucks',
-	to:   'leechba1@gmail.com',
-	from: 'leechbasite',
-	subject: 'test subject',
-	text:    'gmail-send example 1',         // Plain text
-});
-
-function process(req) {
+const BOT_UID = 2;
```
Comment (flga) [inline backend/src/site/legacy-queue.js]: this

File: backend/src/site/main.js [hunk 2]
```diff
@@ -277,7 +277,7 @@ exports.run = function (database) {
   app.use(express.static(legacy));
   const legacyQueueProcessor = require('./legacy-queue');
   app.post('/legacy/queue', async (req, res) => {
-    legacyQueueProcessor.process(req);
+    await legacyQueueProcessor.process(req, database, adminChannel, queueChannel);
```
Comment (JiaRS) [inline backend/src/site/main.js]: what if it fails? should fire up at least a warning

File: backend/src/site/legacy-queue.js [hunk 3]
```diff
@@ -1,34 +1,401 @@
...
+            notes.push("IM");
+        }
+        if (data.enhancer > 0) {
+            notes.push(enhancer.replace(/[{()}]/g, ''));
+        }
+
+        const status = await database.newCustomer(BOT_UID, { // TODO: create bot user
+            date: new Date(),
+            rsn: data.rsn,
+            ba: ba.replace(/[()]/g, ""),
```
Comment (JiaRS) [inline backend/src/site/legacy-queue.js]: points has a space prepended to nm, e.g. " (NM1)"

File: backend/src/database.js [hunk 4]
```diff
@@ -2,9 +2,10 @@ const sqlite3 = require('sqlite3').verbose();
 
 class Database {
     // https://gist.github.com/yizhang82/2ab802f1439490984eb998af3d96b16b
-    constructor() {
+    constructor(path) {
         // this.db = new sqlite3.Database('/usr/leechba/database.db', sqlite3.OPEN_READWRITE, (err) => {
```
Comment (JiaRS) [inline backend/src/database.js]: remove comment

File: backend/src/site/legacy-queue.js [hunk 5]
```diff
@@ -1,37 +1,403 @@
...
+            } else {
+                s2 = "Status: Inserted into queue, please confirm (and remove the unconfirmed note)";
+            }
+
+            return queueChannel.send("```"+ s0 + s1 + s2 + "```");
+        }).then(m => m.pin());
+    }
+
+    // TRIAL REQUEST =================================================================================================================================================
+    if (mail.subject == "trial") {
```
Comment (JiaRS) [inline backend/src/site/legacy-queue.js]: trial requests are useless now, take out

File: backend/src/site/legacy-queue.js [hunk 6]
```diff
@@ -1,37 +1,403 @@
...
+        s0 += "\n\n";
+
+        var notes = ["(unconfirmed)"];
+        if (data.ironman == "yes") {
+            notes.push("IM");
+        }
+        if (data.enhancer > 0) {
+            notes.push(enhancer.replace(/[{()}]/g, ''));
+        }
+        console.log("KINGS", kingsneeded);
```
Comment (JiaRS) [inline backend/src/site/legacy-queue.js]: logging necessary?

File: backend/src/site/legacy-queue.js (hunk 5 above)
Comment (JiaRS) [reply backend/src/site/legacy-queue.js]: actually logging a processed request may be useful but it doesn't log to file

File: backend/src/site/legacy-queue.js [hunk 7]
```diff
@@ -1,37 +1,403 @@
...
+            }
+            if (needH > 0) {
+                needs.push(needH + " heal");
+            }
+            n = needs.join(" + ");
+
+            current += "\n";
+            need += "\n";
+            net += n + "\n";
+            // n = n.replace("att", ":BA_A:");
```
Comment (JiaRS) [inline backend/src/site/legacy-queue.js]: remove comment

File: backend/src/site/legacy-queue.js [hunk 8]
```diff
@@ -1,37 +1,403 @@
...
 }
 
-function sendEmail(subject, body) {
-	send({
-		subject: subject,
-		text: body
-	}, function (err, res) {
-		if (err) {
-			console.log('error sending email');
+async function processQueueReq(mail, database, adminChannel, queueChannel) {
```
Comment (JiaRS) [inline backend/src/site/legacy-queue.js]: take out admin channel

File: backend/src/site/legacy-queue.js [hunk 9]
```diff
@@ -1,37 +1,403 @@
...
+	console.log("processQueueReq", mail);
+    //var channel = this.bot.channels.find("name", this.queueChannel);
+    var month = ('0' + (new Date().getMonth() + 1)).slice(-2);
+    var date = ('0' + new Date().getDate()).slice(-2);
+    var message = "";
+    if (mail.subject == "admin-test") {
```
Comment (JiaRS) [inline backend/src/site/legacy-queue.js]: admin-test is deprecated

File: backend/src/site/legacy-queue.js [hunk 10]
```diff
@@ -1,37 +1,403 @@
...
+        return database.newCustomer(BOT_UID, {
+            date: new Date(),
+            rsn: data.rsn,
+            ba: ba.replace(/[()]/g, "").trim(),
+            services: {
+                bxp: {
+                    [data.skill]: data.amount.split(',').join(''),
+                },
+            },
+            notes: "(unconfirmed)", //TODO: confirmed field
```
Comment (JiaRS) [inline backend/src/site/legacy-queue.js]: TODO

File: backend/src/site/main.js [hunk 11]
```diff
@@ -1,4 +1,4 @@
-exports.run = function (database) {
+exports.run = function (database, adminChannel, queueChannel) {
```
Comment (JiaRS) [inline backend/src/site/main.js]: don't need admin channel

File: backend/src/database.js (hunk 4 above)
Comment (flga) [reply backend/src/database.js]: both of them?

File: backend/src/site/legacy-queue.js (hunk 6 above)
Comment (flga) [reply backend/src/site/legacy-queue.js]: missed that one

File: backend/src/site/legacy-queue.js (hunk 7 above)
Comment (flga) [reply backend/src/site/legacy-queue.js]: was already there

File: backend/src/site/legacy-queue.js (hunk 10 above)
Comment (flga) [reply backend/src/site/legacy-queue.js]: meant it

File: backend/src/database.js (hunk 4 above)
Comment (JiaRS) [reply backend/src/database.js]: Yes please

**Pull request merged**